workos/sso.py

```diff
@@ -248,80 +217,120 @@ def list_connections(
         Returns:
             dict: Connections response from WorkOS.
         """
-        warn(
-            "The 'list_connections' method is deprecated. Please use 'list_connections_v2' instead.",
-            DeprecationWarning,
-        )
-
-        # This method used to accept `connection_type` as a string, so we try
-        # to convert strings to a `ConnectionType` to support existing callers.
-        #
-        # TODO: Remove support for string values of `ConnectionType` in the next
-        #       major version.
-        if connection_type is not None and isinstance(connection_type, str):
-            try:
-                connection_type = ConnectionType[connection_type]
-
-                warn(
-                    "Passing a string value as the 'connection_type' parameter for 'list_connections' is deprecated and will be removed in the next major version. Please pass a 'ConnectionType' instead.",
-                    DeprecationWarning,
-                )
-            except KeyError:
-                raise ValueError("'connection_type' must be a member of ConnectionType")
-
-        if limit is None:
-            limit = RESPONSE_LIMIT
-            default_limit = True
 
         params = {
-            "connection_type": connection_type.value if connection_type else None,
+            "connection_type": connection_type,
             "domain": domain,
             "organization_id": organization_id,
-            "limit": limit,
+            "limit": limit or DEFAULT_RESPONSE_LIMIT,
             "before": before,
             "after": after,
             "order": order or "desc",
         }
 
-        if order is not None:
-            if isinstance(order, Order):
-                params["order"] = str(order.value)
-
-            elif order == "asc" or order == "desc":
-                params["order"] = order
-            else:
-                raise ValueError("Parameter order must be of enum type Order")
-
-        response = self.request_helper.request(
+        response = self._http_client.request(
             "connections",
             method=REQUEST_METHOD_GET,
             params=params,
             token=workos.api_key,
         )
 
-        response["metadata"] = {
-            "params": params,
-            "method": SSO.list_connections,
+        return WorkOsListResource(
+            list_method=self.list_connections,
+            list_args=params,
+            **ListPage[Connection](**response).model_dump(exclude_unset=True),
```

Comment:
I explicitly added `exclude_unset` to the events endpoint because I didn't want to serialize the `before` key, which does not exist in the event list metadata.

Do you think we should extend this to all list endpoints? It might be moot, the API may not omit any fields.